lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php

```diff
@@ -33,6 +33,7 @@ public function addPaymentInfo($paymentName, array $paymentInformation)
             $originAgentBic,
             $paymentInformation['debtorName']
         );
+        $payment->setDueDate($this->createDueDateFromPaymentInformation($paymentInformation));
```

Comment:
I havn't seen the change of behaviour, if the dueDate is not set in the paymentInformation, then the new code will set a date 5 daysin the future, I think this will break some applications...

```suggestion
        if (isset($paymentInformation)) {
            $payment->setDueDate($this->createDueDateFromPaymentInformation($paymentInformation));
        }
```

Reply:
Am I right about this ?

Reply:
Good point. You are right: also for credit transfers, it will set the default date to 5 days in the future, and this may break some applications.

Your suggestion won't work though, because there's a chance `$paymentInformation` is set, while `$paymentInformation['dueDate']` is not.

In the `createDueDateFromPaymentInformation` method itself, there's already a check whether 
`$paymentInformation['dueDate']` is set.

How about setting a default due date in the method `createDueDateFromPaymentInformation`, which can be overridden? I'll make these adjustments so you can have a look.

Reply:
Ah, I missed the check for the field on the payment information. Should read 
```suggestion
    if (isset($aymentInformation['dueDate'])) {
        $payment->setDueDate($this->createDueDateFromPaymentInformation($paymentInformation));
    }
```

I'm going to check your solution as well.

Reply:
Why not hard-code your IBAN into `<CdtrAcct>` for DirectDebitTransfers? ;)